src/extractor/guidance/intersection_generator.cpp

```diff
@@ -158,7 +162,134 @@ Intersection IntersectionGenerator::getConnectedRoads(const NodeID from_node,
+        };
+
+        const auto target_id = GetActualTarget(index);
+        const auto other_target_id = GetActualTarget(other_index);
+
+        const auto coordinate_at_target = node_info_list[target_id];
+        const auto coordinate_at_other_target = node_info_list[other_target_id];
+
+        const auto turn_angle = util::coordinate_calculation::computeAngle(
+            coordinate_at_in_edge, coordinate_at_intersection, coordinate_at_target);
+        const auto other_turn_angle = util::coordinate_calculation::computeAngle(
+            coordinate_at_in_edge, coordinate_at_intersection, coordinate_at_other_target);
+        const double distance_to_target = util::coordinate_calculation::haversineDistance(
+            coordinate_at_intersection, coordinate_at_target);
+
+        const constexpr double MAX_COLLAPSE_DISTANCE = 30;
+        if (distance_to_target < MAX_COLLAPSE_DISTANCE)
+            return false;
+
+        const bool becomes_narrower =
+            angularDeviation(turn_angle, other_turn_angle) < NARROW_TURN_ANGLE &&
+            angularDeviation(turn_angle, other_turn_angle) <
+                angularDeviation(intersection[index].turn.angle,
+                                 intersection[other_index].turn.angle);
+
+        return becomes_narrower;
+    };
+
+    // Only merge valid y-arms
+    if (!isValidYArm(first_index, second_index) || !isValidYArm(second_index, first_index))
+        return false;
+
+    if (angle_between < 60)
+        return true;
+
+    // return false;
+    // Finally, we also allow merging if all streets offer the same name, it is only three roads and
+    // the angle is not fully extreme:
+    if (intersection.size() != 3)
+        return false;
+
+    const std::size_t missing_index = [first_index, second_index]() {
+        if (first_index == 0)
+            return second_index == 2 ? 1 : 2;
+        else if (first_index == 1)
+            return second_index == 2 ? 0 : 2;
+        else
+            return second_index == 1 ? 0 : 1;
+    }();
+
+    // needs to be same road coming in
+    if (node_based_graph.GetEdgeData(intersection[missing_index].turn.eid).name_id !=
+        first_data.name_id)
+        return false;
+
+    // we only allow collapsing of a Y like fork. So the angle to the missing index has to be
+    // roughly equal:
+    const auto y_angle_difference =
+        angularDeviation(angularDeviation(intersection[missing_index].turn.angle,
+                                          intersection[first_index].turn.angle),
+                         angularDeviation(intersection[missing_index].turn.angle,
+                                          intersection[second_index].turn.angle));
+
+    // Allow larger angles if its three roads only of the same name
+    const bool could_be_valid_y_intersection =
+        angle_between < 100 && y_angle_difference < FUZZY_ANGLE_DIFFERENCE;
+    return could_be_valid_y_intersection;
```

Comment:
Why `could_`? This implies that we need some further checks to make this assumptions. Where are they done?